src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -1305,48 +1306,37 @@ codet java_bytecode_convert_methodt::convert_instructions(
       statement=std::string(id2string(statement), 0, statement.size()-2);
     }
 
-    // we throw away the first statement in an exception handler
-    // as we don't know if a function call had a normal or exceptional return
     auto it=method.exception_table.begin();
     for(; it!=method.exception_table.end(); ++it)
     {
       if(cur_pc==it->handler_pc)
       {
-        exprt exc_var=variable(
-          arg0, statement[0],
-          i_it->address,
-          NO_CAST);
-
-        // throw away the operands
-        pop_residue(bytecode_info.pop);
-
-        // add a CATCH-PUSH signaling a handler
-        side_effect_expr_catcht catch_handler_expr;
-        // pack the exception variable so that it can be used
-        // later for instrumentation
-        catch_handler_expr.get_sub().resize(1);
-        catch_handler_expr.get_sub()[0]=exc_var;
-
-        code_expressiont catch_handler(catch_handler_expr);
-        code_labelt newlabel(label(std::to_string(cur_pc)),
-                             code_blockt());
-
-        code_blockt label_block=to_code_block(newlabel.code());
-        code_blockt handler_block;
-        handler_block.move_to_operands(c);
-        handler_block.move_to_operands(catch_handler);
-        handler_block.move_to_operands(label_block);
-        c=handler_block;
-        break;
+        // at the beginning of a handler, clear the stack and
+        // push the corresponding exceptional return variable
+        stack.clear();
+        auxiliary_symbolt new_symbol;
+        new_symbol.is_static_lifetime=true;
+        int file_name_length=
+          id2string(method.source_location.get_file()).size();
+        // remove the file extension
+        const std::string &file_name=
+          id2string(method.source_location.get_file()).
+             substr(0, file_name_length-5);
+        // generate the name of the exceptional return variable
+        const std::string &exceptional_var_name=
```

Comment:
This creates the same exception variable name for two different methods using the following code: 

```java
class Ex extends Throwable {}

class A {
  public static void f() {
    try {
      throw new Ex();
    }
    catch(Ex ex) { }
  }
}

class B {
  public static void f() {
    try {
      throw new Ex();
    }
    catch(Ex ex) { }
  }
}

public class test {
 public static void main (String arg[]) {
   A.f();
   B.f();
 }
}
```

Perhaps consider making the `class_symbol` available to `convert_instructions` and use its name? E.g. pass it as a parameter or make it a field of `java_bytecode_convert_methodt`?